actions/watch.ts

```diff
@@ -24,28 +24,42 @@ export const checkForAndPlaceOrder: ActionFn = async (
       chainContext.provider
     );
     try {
-      const order = await contract.getTradeableOrder();
-      const signature = contract.interface.encodeFunctionResult(
-        "getTradeableOrder()",
-        [Array.from(order)]
-      );
-
-      const orderIsValid = order.validTo * 1000 > new Date().getTime()
-
-      if (orderIsValid) {
-        console.log(`Placing Order: ${order}`);
-        await placeOrder(
-          { ...order, from: contract_address, signature },
-          chainContext.api_url
+      const [isCancelled, endTime]: [string, string] = await Promise.allSettled([contract.cancelled(), contract.endTime()]).then(([cancelledResult, endTimeResult])=> {
+        if(cancelledResult.status === "rejected" || endTimeResult.status === "rejected") {
+          throw "Rejected result"
+        }
+
+        return [cancelledResult.value, endTimeResult.value]
+      })
+
+      //Give it a few more 15 minutes so it reproduces the last order
+      const _15_MINUTES = 900
+      const endTimeIsValid = (parseInt(endTime, 10) + _15_MINUTES) * 1000 > new Date().getTime()
```

Comment:
can you give more details on this?

Reply:
What kind of detail? This is to check if endTime is still valid.

Reply:
Gave it a diff name